crates/biome_diagnostics/src/display/frame.rs

```diff
@@ -254,6 +254,76 @@ pub(super) fn print_frame(fmt: &mut fmt::Formatter<'_>, location: Location<'_>)
     fmt.write_str("\n")
 }
 
+pub(super) fn print_highlighted_frame(
+    fmt: &mut fmt::Formatter<'_>,
+    location: Location<'_>,
+) -> io::Result<()> {
+    let Some(span) = location.span else {
+        return Ok(());
+    };
+    let Some(source_code) = location.source_code else {
+        return Ok(());
+    };
+
+    let source = SourceFile::new(source_code);
+
+    let start = source.location(span.start())?;
+    let end = source.location(span.end())?;
```

Comment:
Ah, this makes me realize we need to calculate line numbers for every match, which can potentially be a bit expensive. No big deal either way, but a good reminder that I may want to see if the Grit engine can already report the line numbers for us. At least it tracks them for some purposes already (though not everywhere), so I may be able to pull those out.

Reply:
Would you like me to add a TODO to keep track of this possibility?

Reply:
Would be appreciated!